linear_algebra/src/lib.py

```diff
@@ -183,6 +182,16 @@ def angle(self, other: Vector, deg: bool = False) -> float:
         else:
             return math.acos(num / den)
 
+    def __eq__(self, vector: object) -> bool:
```

Comment:
Could you move the method higher up in the file, before all the public methods? Having this one dunder method within all the public methods, rather than having it alongside all the other dunder methods, makes it harder to find.